ITGA7.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import annotations
+import os
+import sys
+from Bio import SeqIO, SeqUtils, SeqRecord
+from Bio.SeqUtils import gc_fraction
+
+
+class BiologicalSequence:
+    """
+    Represents a generic biological sequence.
+    Attributes:
+    - sequence (str): The biological sequence.
+    Methods:
+    - __len__(): Returns the length of the sequence.
+    - __getitem__(index: int): Returns the character at the specified index.
+    - __str__(): Returns a string representation of the sequence.
+    - is_valid_alphabet(): Checks if the sequence uses a valid alphabet.
+    """
+
+    def __init__(self, sequence: str):
+        self.sequence = sequence
+
+    def __len__(self) -> int:
+        return len(self.sequence)
+
+    def __getitem__(self, index: int) -> str:
+        return self.sequence[index]
+
+    def __str__(self) -> str:
+        return f"{self.sequence}"
+
+    def is_valid_alphabet(self) -> bool:
+        """
+        Checks if the sequence uses a valid alphabet.
+        """
+        try:
+            unique_chars = set(self.sequence)
+            return unique_chars <= self.ALPHABET
+        except AttributeError as e:
+            raise NotImplementedError(
+                "Is valid alphabet method not implemented for this class."
+            ) from e
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    Represents a nucleic acid sequence.
+    Methods:
+    - complement(): Returns the complemented sequence.
+    - gc_content(percentage: bool = True) -> int | float: Calculates the GC content of the sequence.
+    Raises:
+    - NotImplementedError: If complement method is not implemented for this class.
+    """
+
+    def complement(self) -> DNASequence | RNASequence:
+        try:
+            complemented_sequence = "".join(
+                [self.COMPLEMENT_MAP[base] for base in self.sequence]
+            )
+            return self.__class__(complemented_sequence)
+        except AttributeError as e:
+            raise NotImplementedError(
+                "Complement method not implemented for this class."
+            ) from e
+
+    def gc_content(self, percentage: bool = True) -> int | float:
+        """
+        Calculates the GC content of the sequence.
+        Args:
+        - percentage (bool): If True, returns the result as a percentage.
+        Returns:
+        - int | float: GC content value.
+        """
+        gc_symbols = set("GCgc")
+        gc_count = sum(1 for nucleotide in self.sequence if nucleotide in gc_symbols)
```

Comment:
можно и так, но можно и сяк:
```suggestion
        gc_count = (self.sequence.count('G') + 
                    self.sequence.count('C') + 
                    self.sequence.count('g') + 
                    self.sequence.count('c'))

```

Reply:
ну или вообще привести все к одному регистру локально с помощью .upper(), что выглядит тоже здраво

Reply:
Так зато все считается за один прогон по строке. Думаю это не так существенно, один вариант чуть эффективнее, другой - чуть читаемее. Кому что больше нравится.